build_modules/lib/src/module_builder.dart

```diff
@@ -122,14 +105,14 @@ class ModuleBuilder implements Builder {
     var assetToPrimary = await buildStep.fetchResource(_assetToPrimary);
     var readMetas = await buildStep.fetchResource(_readMetas);
     var primaryToClean = await buildStep.fetchResource(_primaryToClean);
-    var asset =
+    var cleanMetaAsset =
         new AssetId(buildStep.inputId.package, 'lib/$metaModuleCleanExtension');
     // If we can't read the clean meta module it is likely that this package
     // is in a module cycle so fall back to the fine strategy.
-    if (await buildStep.canRead(asset) && _isCoarse) {
-      if (!readMetas.contains(asset)) {
-        await _processMeta(
-            buildStep, asset, primaryToClean, assetToPrimary, readMetas);
+    if (await buildStep.canRead(cleanMetaAsset) && _isCoarse) {
```

Comment:
Discussed offline. The issue that the file was getting created is because of this conditional. I will follow up with a PR that reverses the check.